capstone-backend/app.js

```diff
@@ -1,49 +1,39 @@
-var express = require('express');
-var path = require('path');
-var logger = require('morgan');
-var cookieParser = require('cookie-parser');
-var bodyParser = require('body-parser');
-
-var callback = require('./routes/callback');
-
-var app = express();
-
-// view engine setup
-app.set('views', path.join(__dirname, 'views'));
-app.set('view engine', 'ejs');
-
-// uncomment after placing your favicon in /public
-//app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
-app.use(logger('dev'));
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(cookieParser());
-app.use(express.static(path.join(__dirname, 'public')));
-
-// app.use('/', index);
-app.use(express.static(path.join(__dirname, 'client/build')));
-app.use('/callback', callback);
-
-app.get('/*', (req, res) => {
-  res.sendFile(path.join(__dirname, 'client/build', 'index.html'))
+const express = require('express')
+const morgan = require('morgan')
+const cors = require('cors')
+const Parse = require('parse/node')
+const {PARSE_APP_ID, PARSE_JAVASCRIPT_KEY} = require('./config')
+
+const app = express()
+const port = process.env.PORT || 3001
+
+app.use(express.json())
+app.use(morgan("tiny"))
+app.use(cors())
+
+Parse.initialize(PARSE_APP_ID, PARSE_JAVASCRIPT_KEY)
+Parse.serverURL = "https://parseapi.back4app.com"
+
+app.post('/register', async (req, res) => {
+  let user = new Parse.User(req.body)
+
+  try {
+      await user.signUp()
+      res.status(201)    
+      res.send({"user" : user})
+  } catch (error) {
+      res.status(400)
+      res.send({"error" : "Failed to create user: " + error })
+  }
 })
 
-// catch 404 and forward to error handler
-app.use(function(req, res, next) {
-  var err = new Error('Not Found');
-  err.status = 404;
-  next(err);
-});
-
-// error handler
-app.use(function(err, req, res, next) {
-  // set locals, only providing error in development
-  res.locals.message = err.message;
-  res.locals.error = req.app.get('env') === 'development' ? err : {};
-
-  // render the error page
-  res.status(err.status || 500);
-  res.render('error');
-});
-
+app.post('/login', async (req, res) => {
+  try {
+    const user = await Parse.User.logIn(req.body.username, req.body.password)
```

Comment:
Yay! const! 👍

Reply:
Nit: 
If we find time later, let's go over considering what happens if `username` or `password` are absent from the `req.body`